### ISSUE TITLE
Implementing Interfaces

### ISSUE THREAD

**Issue by aluanhaddad** [pgenfer/mixinSharp#11]
First of all great work on this project!

Now for some ideas.
 
Say I have the following mixin
```C#
public class NameMixin: INamed
{
    public string Name { get; set; }
}
```
and ```INamed``` is defined as follows
```C#
public interface INamed
{
    string Name { get; set; } = string.Empty;
}
```
Furthermore, I have the following class
```C#
public class Person
{
    NameMixin _name = new NameMixin();
}
```
When I mix the ```NameMixin``` class into my type, I would like the option for the type to be further augmented by having the ```INamed``` interface automatically added to its interface list. Conversely, it would be very convenient to offer an _Implement with NamedMixin_ code action when adding the ```INamed``` interface to a type not having an implementation.

Part of the reason I want this functionality comes from experience with languages which have mixins. Let's take Scala as an example. Scala uses traits to declare mixins using the syntax
```scala
case object Person with Named
```
with the trait Named being defined as
```scala
trait Named { 
  var name = "" // this is terrible Scala, but it's just mean to be illustrative
}
```
The generated class files actually contain an interface and a class implementing that interface which together make up the ```Named``` trait. The point is we are augmenting our API surface transparently via delegation. Other languages follow this pattern as well, so it feels natural. More philosophically, I think mixins are a natural way to provide cross-axial classification, not just delegating implementation. Personally, I always use as few concrete types in my public APIs as is reasonable. 

Anyway, these are just some thoughts. 

Again, thanks for working on this project and sharing it with the Roslyn community!

**Comment by pgenfer** [comment]
Hi aluanhaddad,
Thanks a lot for your feedback and your suggestions!

I think these are two very good ideas, and at a first sight, the implementation effort to add these features should really be manageable.

I'm currently finishing some features for V1.2, but after that I will check if I can integrate your suggestions in a further version (maybe 1.3).

**Comment by aluanhaddad** [comment]
@pgenfer Awesome! I'm very much looking forward to it :grinning:

**Comment by pgenfer** [comment]
Hi!
As these are actually two different features,  I just separated them into two issues (#13 and #14).
If it's ok I would close this issue here since it is replaced by the two other ones now.  
(#13 should be implemented in V1.3 which might be released during this weekend).

**Comment by aluanhaddad** [comment]
It definitely makes sense to split these up.
